### ISSUE TITLE
Request for future xmage updates : improve offline game

### ISSUE THREAD

**Issue by WildstarDerek76** [magefree/mage#12999]
First of all, I thank those who, by dedicating themselves to the development of this project, let the passion behind this immense and excellent work pass. Having said this, I would like to forward my request hoping it will be fulfilled for the next versions of xmage : I am an xmage player who plays mainly offline, I wonder if it will be possible to see in the next versions an expansion for offline play with more choices for bot intelligence to face and if the capabilities of the opponent driven by the CPU will be more complex and developed to make offline play more challenging and as close as possible to a game with human players. Thank you.

**Comment by jeffwadsworth** [comment]
The MAD AI is quite capable.  Behold.  https://youtu.be/_SGbinPzLcc

**Comment by JayDi85** [comment]
AI experience depends on used deck and cards. Some can be good for AI: simple actions and targets, clear good/bad effects, etc. But some can be bad: complex combos with discard, X values, sacrifice, etc.

**Comment by WildstarDerek76** [comment]
> AI experience depends on used deck and cards. Some can be good for AI: simple actions and targets, clear good/bad effects, etc. But some can be bad: complex combos with discard, X values, sacrifice, etc.

Thank you for your answer, so in this aspect in xmage it will not be possible to see in future updates a development regarding offline play with an improved AI in handling more complex game operations? Thank you for your great work and your eventual response.

**Comment by JayDi85** [comment]
There are many places to improve. Example: attack/block decisions (current version do not use simulation of all possible combinations for better choice). Same for yes/no and other choices on resolve. And more: #10197

**Comment by WildstarDerek76** [comment]
> There are many places to improve. Example: attack/block decisions (current version do not use simulation of all possible combinations for better choice). Same for yes/no and other choices on resolve. And more: #10197

I agree with u Jaydi85, clearly improving the AI responses to assist in more complex plays is not easy I suppose, my question was more related to the desire to improve this aspect to make the challenges more satisfying with an ai capable of developing more complex combos as you mentioned. That's all, or if offline development was seen as a more marginal aspect of the project.

**Comment by sprangg** [comment]
Currently single player offline play is a marginal aspect of XMage, unless someone comes along and starts focusing on that.

In the meantime, you might have some luck with [Forge](https://github.com/Card-Forge/forge), which is the other open source mtg engine that's very much focused on offline play instead of online play.

**Comment by JayDi85** [comment]
AI improves time by time with new releases, so it’s better to report more specific cards or bad use cases.
